File: api/csharp/ToolGood.TextFilter.Api.ConsulTest/Tests/ConsulTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PetaTest;
using ToolGood.TextFilter.Api;

namespace ToolGood.TextFilter.Test.Tests
{
    [TestFixture]
    public class ConsulTest
    {

        [Test]
        public void GetServerName()
        {
            var urlsTask = TextFilterConfig.Instance.GetServiceUrls( ServiceUrlType.Http);
            urlsTask.Wait();
            var urls = urlsTask.Result;
            Assert.Greater(urls.Count, 0);
            Assert.AreEqual("http://127.0.0.1:9191", urls[0]);
        }

        [Test]
        public void Test_Http()
        {
            var provider = TextFilterConfig.Instance.CreateTextFilterProvider();
            var result = provider.TextFilter(new Api.Datas.Requests.TextFilterRequest("abb"));
            Assert.AreEqual(result.Code, 0);
        }

        [Test]
        public void Test_Grpc()
        {
            var provider = TextFilterConfig.Instance.CreateTextFilterGrpcClient();
            var result = provider.TextFilter(new Api.Datas.Requests.TextFilterRequest("abb"));
            Assert.AreEqual(result.Code, 0);
        }



    }
}

File: api/java/toolgood.textfilter.api/src/test/java/toolgood/textfilter/api/Tests/ConsulTest.java
package toolgood.textfilter.api.Tests;

import static org.junit.Assert.assertEquals;

import java.util.List;

import org.junit.Test;

import toolgood.textfilter.api.ServiceUrlType;
import toolgood.textfilter.api.TextFilterConfig;
import toolgood.textfilter.api.Datas.Requests.TextFilterRequest;
import toolgood.textfilter.api.Datas.Texts.TextFilterResult;
import toolgood.textfilter.api.Interfaces.ITextFilterProvider;

public class ConsulTest {

    @Test
    public void GetServerName() {
        TextFilterConfig config = TextFilterConfig.Instance();

        List<String> urls = config.GetServiceUrls(ServiceUrlType.Http);
        assertEquals(1, urls.size());
        assertEquals("http://127.0.0.1:9191", urls.get(0));
    }

    @Test
    public void Test_Http() {
        ITextFilterProvider provider = TextFilterConfig.Instance().CreateTextFilterProvider();
        TextFilterResult result = provider.TextFilter(new TextFilterRequest("abb", false, false));
        assertEquals(result.code, 0);
    }

    @Test
    public void Test_Grpc() {
        ITextFilterProvider provider = TextFilterConfig.Instance().CreateTextFilterGrpcClient();
        TextFilterResult result = provider.TextFilter(new TextFilterRequest("abb", false, false));
        assertEquals(result.code, 0);
    }

}
